src/commands/test.js

```diff
@@ -35,48 +37,56 @@ module.exports = {
       return
     }
 
-    const platform = getPlatform();
+    const platform = getPlatform(logs)
     if (!version) {
-      let result = await fetch('https://api.github.com/repos/LimeChain/matchstick/releases/latest');
-      let json = await result.json();
-      version = json.tag_name;
+      let result = await fetch('https://api.github.com/repos/LimeChain/matchstick/releases/latest')
+      let json = await result.json()
+      version = json.tag_name
     }
 
-    const url = `https://github.com/LimeChain/matchstick/releases/download/${version}/${platform}`;
+    const url = `https://github.com/LimeChain/matchstick/releases/download/${version}/${platform}`
 
-    let binary = new Binary(platform, url, version);
-    await binary.install(force);
-    datasource ? binary.run(datasource) : binary.run();
+    if (logs) {
+      console.log(`Download link: ${url}`)
+    }
+
+    let binary = new Binary(platform, url, version)
+    await binary.install(force)
+    datasource ? binary.run(datasource) : binary.run()
   }
 }
 
-function getPlatform() {
-  const type = os.type();
-  const arch = os.arch();
-  const release = os.release();
-  const cpuCore = os.cpus()[0];
-  const majorVersion = semver.major(release);
-  const isM1 = cpuCore.model.includes("Apple M1");
+function getPlatform(logs) {
+  const type = os.type()
+  const arch = os.arch()
+  const release = os.release()
+  const cpuCore = os.cpus()[0]
+  const majorVersion = semver.major(release)
+  const isM1 = cpuCore.model.includes("Apple M1")
+
+  if (logs) {
+    console.log(`OS type: ${type}\nOS arch: ${arch}\nOS release: ${release}\nOS major version: ${majorVersion}\nCPU model: ${cpuCore.model}`)
```

Comment:
Here you can even remove the `\n`s because the back ticks are sensitive to line breaks:

```javascript
console.log(`OS type: ${type}
OS arch: ${arch}
OS release: ${release}
OS major version: ${majorVersion}
CPU model: ${cpuCore.model}`)
```

But that's only preference, choose whatever you prefer, I'm just mentioning the possibility 🙂

Reply:
Yeah, I tried it just now but it also factors in indentation which either means that all the logs will have weird indentation in the console or they will have 0 indentation in the code which will also look weird imo.